File: app/src/main/java/com/exemple/habitapp/ReminderReceiver.java
package com.exemple.habitapp;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class ReminderReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String type = intent.getStringExtra(ReminderScheduler.EXTRA_TYPE);
        if (type == null) {
            type = ReminderScheduler.TYPE_ROUTINE;
        }
        NotificationHelper.showReminder(context, type);
    }
}

File: app/src/main/java/com/exemple/habitapp/notifications/ReminderReceiver.kt
package com.exemple.habitapp.notifications

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class ReminderReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent?) {
        val type = intent?.getStringExtra(ReminderScheduler.EXTRA_TYPE).orEmpty()
        val prefs = context.getSharedPreferences("habit_data", Context.MODE_PRIVATE)
        if (!prefs.getBoolean("notifications_enabled", true)) return
        if (type == "water" && !prefs.getBoolean("water_reminder_enabled", true)) return
        if (type == "focus" && !prefs.getBoolean("focus_reminder_enabled", true)) return
        NotificationHelper.showReminder(context, type.ifBlank { "routine" })
        ReminderScheduler.scheduleDefaultReminders(context)
    }
}
